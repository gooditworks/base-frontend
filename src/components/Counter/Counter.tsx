import {FC} from "react"

interface Props {
  value: number
  onIncrement: () => void
  onDecrement: () => void
}

const Counter: FC<Props> = ({value, onIncrement, onDecrement}: Props) => (
  <div className="flex items-center space-x-8 text-5xl">
    <button onClick={onDecrement} type="button" data-test="counter_dec">
      -
    </button>
    <span data-test="counter_value">{value}</span>
    <button onClick={onIncrement} type="button" data-test="counter_inc">
      +
    </button>
  </div>
)

export default Counter
