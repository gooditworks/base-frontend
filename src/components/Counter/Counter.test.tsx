import {render} from "@testing-library/react"

import Counter from "./Counter"

it("matches snapshot", () => {
  const counter = <Counter value={0} onIncrement={() => {}} onDecrement={() => {}} />
  const {container} = render(counter)

  expect(container.firstChild).toMatchSnapshot()
})
