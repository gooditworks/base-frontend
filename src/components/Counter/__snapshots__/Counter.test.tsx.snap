// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="flex items-center space-x-8 text-5xl"
>
  <button
    data-test="counter_dec"
    type="button"
  >
    -
  </button>
  <span
    data-test="counter_value"
  >
    0
  </span>
  <button
    data-test="counter_inc"
    type="button"
  >
    +
  </button>
</div>
`;
